packages/core/src/components/section/section.tsx

```diff
@@ -201,17 +199,30 @@ export const Section: React.FC<SectionProps> = React.forwardRef((props, ref) =>
                             )}
                         </div>
                     </div>
-                    {isHeaderRightContainerVisible && (
-                        <div className={Classes.SECTION_HEADER_RIGHT}>
-                            {rightElement}
-                            {collapsible &&
-                                (isCollapsed ? (
-                                    <ChevronDown className={Classes.TEXT_MUTED} />
-                                ) : (
-                                    <ChevronUp className={Classes.TEXT_MUTED} />
-                                ))}
-                        </div>
-                    )}
+
+                    {collapsible &&
+                        (isCollapsed ? (
+                            <ChevronDown className={Classes.TEXT_MUTED} />
+                        ) : (
+                            <ChevronUp className={Classes.TEXT_MUTED} />
+                        ))}
+                </div>
+            )}
+
+            {rightElement && (
+                // Needs to be outside of the section header because section header may be a button (when collapsable), and
+                // interactive elements (that may be rightElement) can't be inside other buttons.
+                // Use an overlay to position rightElement absolutely in the header.
+                <div className={Classes.SECTION_HEADER_OVERLAY}>
+                    <div
+                        className={Classes.SECTION_HEADER_RIGHT}
+                        style={{
+                            // 25px accounts for the collapse chevron
+                            marginRight: collapsible ? 25 : undefined,
```

Comment:
Could there be any issues with this element overlapping other elements (I think just title/subtitle)? It's possible we'd need to do something to visually take up the space.